### PR TITLE
ci: update release ci to use macOS

### PR DESCRIPTION
**Pull request by JeyJeyGao**
The binaries for macOS cannot be executed without macOS codesign, so optimize the pipeline to include codesign.

- Updated CI to build binaries on Linux and codesign on macOS
- Use Github Artifact store temporary files between jobs. The retention time is 1 day.

Example release: https://github.com/JeyJeyGao/notation-azure-kv/releases/tag/v1.0.46
Example action: https://github.com/JeyJeyGao/notation-azure-kv/actions/runs/4948193196

Test:
- Linux x64 bianry
- Windows x64 binary
- macOS arm64 bianry
- macOS x64 bianry

Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ permissions:
 jobs:
   build:
     name: Release
-    runs-on: ubuntu-latest
+    runs-on: macos-13
```
Comment (shizhMSFT) [inline .github/workflows/release.yml:15]: Could we just use mac to do codesign or just use mac to build mac release but not others?

Comment (JeyJeyGao) [reply .github/workflows/release.yml]: Updated to build on linux and codesign on macOS.

File: scripts/release.sh [hunk 2]
```diff
@@ -10,57 +10,14 @@ fi
...
-
-    (cd "${artifacts_dir}" && sha256sum "$(basename "${artifact_name}")" >>"${checksum_name}")
-
-    # Add the artifact to the list
-    artifacts+=("${artifact_name}")
+# create checksums
+mapfile -t artifacts < <(find "$artifacts_dir" -type f)
+for artifact in "${artifacts[@]}"; do
+    (cd "${artifacts_dir}" && shasum -a 256 "$(basename "${artifact}")" >>"${checksum_name}")
 done
```
Comment (shizhMSFT) [inline scripts/release.sh:21]: It seems too complex. Will something like `sha256sum *` work?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -11,18 +12,90 @@ permissions:
...
+      - name: Download osx-x64 artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: osx-x64
+          path: ${{ github.workspace }}/bin/artifacts
+      - name: Download osx-arm64 artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: osx-arm64
+          path: ${{ github.workspace }}/bin/artifacts
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply .github/workflows/release.yml]: Updated.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -11,18 +12,90 @@ permissions:
...
+      - name: Upload ${{ matrix.runtime }} artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.runtime }}
+          path: ${{ github.workspace }}/bin/artifacts
+          retention-days: 1
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    needs: [macos_codesign]
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply .github/workflows/release.yml]: Updated.

File: scripts/release.sh (hunk 2 above)
Comment (JeyJeyGao) [reply scripts/release.sh]: The artifacts in different directories, so I think we cannot.

Comment (shizhMSFT) [reply scripts/release.sh]: I think we can move files around, can't we?

Comment (JeyJeyGao) [reply scripts/release.sh]: Updated to move artifacts to `release_assets` directory then generate checksum

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**